okapi-core/src/main/java/org/folio/okapi/managers/TimerManager.java

```diff
@@ -0,0 +1,312 @@
+                });
+            seq++;
+          }
+        }
+      }
+      return future;
+    });
+  }
+
+  /**
+   * Fire a timer - invoke module.
+   * @param tenantId tenant identifier
+   * @param md module descriptor of module to invoke
+   * @param timerDescriptor timer descriptor in use
+   */
+  private void fireTimer(String tenantId, ModuleDescriptor md, TimerDescriptor timerDescriptor) {
+    RoutingEntry routingEntry = timerDescriptor.getRoutingEntry();
+    String path = routingEntry.getStaticPath();
+    HttpMethod httpMethod = routingEntry.getDefaultMethod(HttpMethod.POST);
+    ModuleInstance inst = new ModuleInstance(md, routingEntry, path, httpMethod, true);
+    MultiMap headers = MultiMap.caseInsensitiveMultiMap();
+    logger.info("timer {} call start module {} for tenant {}",
+        timerDescriptor.getId(), md.getId(), tenantId);
+    proxyService.callSystemInterface(headers, tenantId, inst, "")
+        .onFailure(cause ->
+            logger.info("timer call failed to module {} for tenant {} : {}",
+                md.getId(), tenantId, cause.getMessage()))
+        .onSuccess(res ->
+            logger.info("timer call succeeded to module {} for tenant {}",
+                md.getId(), tenantId));
+  }
+
+  Future<ModuleDescriptor> getModuleForTimer(String tenantId, String timerId) {
+    return tenantManager.getEnabledModules(tenantId).compose(list -> {
+      String product = timerId.substring(0, timerId.indexOf(TIMER_ENTRY_SEP));
+      for (ModuleDescriptor md : list) {
+        if (product.equals(md.getProduct())) {
+          return Future.succeededFuture(md);
+        }
+      }
+      return Future.succeededFuture(null);
+    }).recover(cause -> Future.succeededFuture(null));
+  }
+
+  /**
+   * Handle a timer timer.
+   *
+   * <p>This method is called for each timer in each tenant and for each instance in
+   * the Okapi cluster. We have no way of stopping a timer.. So in fact this call
+   * may result in nothing.. because the timer descriptor is obsolete (patch or module update)
+   * @param tenantId tenant identifier
+   * @param timerDescriptor descriptor that this handling
+   */
+  private void handleTimer(String tenantId, TimerDescriptor timerDescriptor) {
+    logger.info("timer {} handle for tenant {}", timerDescriptor.getId(), tenantId);
+    final String timerId = timerDescriptor.getId();
+    tenantTimers.get(tenantId, timerId)
+        .compose(currentDescriptor -> {
+          // if value has changed, then stop ..
+          if (!isSimilar(timerDescriptor, currentDescriptor)) {
+            return Future.succeededFuture();
+          }
+          // this timer is latest and current .. do the work..
+          // find module for this timer.. If module is not found, it was disabled
+          // in the meantime and timer is stopped.
+          return getModuleForTimer(tenantId, timerId).compose(md -> {
+            if (md == null) {
+              timerRunning.remove(timerId);
```

Comment:
If one tenant disable a timer modules, what does this `remove` mean for other tenants that still have the same time module enabled?

Reply:
Must be tenant specific